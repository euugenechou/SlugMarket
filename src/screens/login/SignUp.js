// React imports 
import React from "react";
import {
  TextInput,
  StyleSheet,
  Text,
  View,
  Alert,
  ScrollView
} from "react-native";
import Icon from "react-native-vector-icons/Entypo";
import { Button } from "react-native-elements";
// AWS imports 
import { Auth } from "aws-amplify";
// Local imports
import TopAlert from "../components/TopAlert";

export default class App extends React.Component {
  state = {
    name: "",
    email: "",
    phone_number: "",
    password: "",
    confirmationCode: "",
    visible: false
  };

  static navigationOptions = () => ({
    headerTintColor: "teal",
    headerStyle: {
      height: 40,
      backgroundColor: "white",
      borderBottomWidth: 0
    }
  });

  onChangeText(key, value) {
    this.setState({ [key]: value });
  }

  signUp() {
    Auth.signUp({
      username: this.state.email,
      password: this.state.password,
      attributes: {
        name: this.state.name,
        phone_number: this.state.phone_number
      }
    })
      .then(() =>
        this.props.navigation.navigate("ConfirmScreen", {
          userName: this.state.email
        })
      )
      .catch(() =>
        this.setState({
          visible: true,
          errorMessage: "Username, password, phone number, or email incorrect"
        })
      );
  }

  render() {
    return (
      <ScrollView centerContent={true} contentContainerStyle={styles.container}>
        <TopAlert
          visible = { this.state.visible }
          message = { this.state.errorMessage }
        />
        <Icon
          name = "chevron-small-left"
          size = { 45 }
          color = "teal"
          style = { {paddingVertical: 10} }
          onPress = { () => this.props.navigation.navigate("SignInScreen") }
        />
        <View style = { styles.container }>
          <Text style = { styles.text }> First Name </Text>
          <TextInput
            onChangeText = { value => this.onChangeText("name", value) }
            style = { styles.input }
            placeholder = "e.g.: Sammy"
            placeholderTextColor = "gray"
            autoCapitalize = "none"
            autoCorrect = { false }
          />
          <Text style = { styles.text }> Password </Text>
          <TextInput
            onChangeText = { value => this.onChangeText("password", value) }
            style = { styles.input }
            secureTextEntry = { true }
            placeholder=  "e.g.: Slug6969"
            placeholderTextColor = "gray"
            autoCapitalize = "none"
            autoCorrect = { false }
          />
          <Text style = { styles.text }> Phone Number </Text>
          <TextInput
            onChangeText = { value => this.onChangeText("phone_number", value) }
            style = { styles.input }
            placeholder = "e.g.: +15101234567"
            placeholderTextColor = "gray"
            autoCapitalize = "none"
            autoCorrect = { false }
          />
          <Text style = { styles.text }> UCSC Email </Text>
          <TextInput
            onChangeText = { value => this.onChangeText("email", value) }
            style = { styles.input }
            placeholder = "e.g.: sammyslug@ucsc.edu"
            placeholderTextColor = "gray"
            keyboardType = "email-address"
            autoCapitalize = "none"
            autoCorrect = {false}
          />
          <Button
            raised
            color = "white"
            title = "Sign Up"
            fontWeight = "bold"
            backgroundColor = "teal"
            borderRadius = { 5 }
            containerViewStyle = { styles.buttonContainer }
            onPress = { () => this.signUp() }
          />
        </View>
      </ScrollView>
    );
  }
}

const styles = StyleSheet.create({
  input: {
    justifyContent: "center",
    alignSelf: "center",
    height: 50,
    color: "black",
    borderBottomWidth: 0.7,
    borderBottomColor: "darkgray",
    marginBottom: 20,
    width: 300
  },
  container: {
    flex: 1,
    backgroundColor: "white",
    paddingVertical: 10
  },
  buttonContainer: {
    width: 300,
    paddingTop: 10,
    alignSelf: "center"
  },
  text: {
    paddingLeft: 36,
    fontSize: 20,
    color: "black",
    fontWeight: "700",
    textAlign: "left",
    paddingTop: 10
  }
});
