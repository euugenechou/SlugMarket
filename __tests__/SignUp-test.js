// __tests__/SignUp-test.js
import 'react-native';
import React from 'react';
import SignUp from '../src/screens/login/SignUp';

import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <SignUp />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});