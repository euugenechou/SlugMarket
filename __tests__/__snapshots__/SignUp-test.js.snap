// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  accessible={false}
  centerContent={true}
  contentContainerStyle={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "paddingVertical": 10,
    }
  }
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "teal",
            "fontSize": 45,
          },
          Object {
            "paddingVertical": 10,
          },
          Object {
            "fontFamily": "Entypo",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      ï……
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "flex": 1,
          "paddingVertical": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "black",
            "fontSize": 20,
            "fontWeight": "700",
            "paddingLeft": 32,
            "paddingTop": 10,
            "textAlign": "left",
          }
        }
      >
         First Name 
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        onChangeText={[Function]}
        placeholder="e.g.: Sammy"
        placeholderTextColor="gray"
        style={
          Object {
            "alignSelf": "center",
            "borderBottomColor": "darkgray",
            "borderBottomWidth": 0.7,
            "color": "black",
            "height": 50,
            "justifyContent": "center",
            "marginBottom": 20,
            "width": 300,
          }
        }
        underlineColorAndroid="transparent"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "black",
            "fontSize": 20,
            "fontWeight": "700",
            "paddingLeft": 32,
            "paddingTop": 10,
            "textAlign": "left",
          }
        }
      >
         Password 
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        onChangeText={[Function]}
        placeholder="e.g.: Slug1965"
        placeholderTextColor="gray"
        secureTextEntry={true}
        style={
          Object {
            "alignSelf": "center",
            "borderBottomColor": "darkgray",
            "borderBottomWidth": 0.7,
            "color": "black",
            "height": 50,
            "justifyContent": "center",
            "marginBottom": 20,
            "width": 300,
          }
        }
        underlineColorAndroid="transparent"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "black",
            "fontSize": 20,
            "fontWeight": "700",
            "paddingLeft": 32,
            "paddingTop": 10,
            "textAlign": "left",
          }
        }
      >
         Phone Number 
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        onChangeText={[Function]}
        placeholder="e.g.: +15101234567"
        placeholderTextColor="gray"
        style={
          Object {
            "alignSelf": "center",
            "borderBottomColor": "darkgray",
            "borderBottomWidth": 0.7,
            "color": "black",
            "height": 50,
            "justifyContent": "center",
            "marginBottom": 20,
            "width": 300,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "black",
            "fontSize": 20,
            "fontWeight": "700",
            "paddingLeft": 32,
            "paddingTop": 10,
            "textAlign": "left",
          }
        }
      >
         UCSC Email 
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        keyboardType="email-address"
        onChangeText={[Function]}
        placeholder="e.g.: sammyslug@ucsc.edu"
        placeholderTextColor="gray"
        style={
          Object {
            "alignSelf": "center",
            "borderBottomColor": "darkgray",
            "borderBottomWidth": 0.7,
            "color": "black",
            "height": 50,
            "justifyContent": "center",
            "marginBottom": 20,
            "width": 300,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "marginLeft": 15,
              "marginRight": 15,
            },
            undefined,
            Object {
              "alignSelf": "center",
              "paddingTop": 10,
              "width": 300,
            },
            Object {
              "borderRadius": 5,
            },
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
        >
          <View
            pointerEvents="box-only"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#9E9E9E",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 19,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "backgroundColor": "teal",
                },
                Object {
                  "borderRadius": 5,
                },
                Object {
                  "padding": 12,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 20,
                    },
                    Object {
                      "color": "white",
                    },
                    Object {
                      "fontSize": 17.5,
                    },
                    undefined,
                    undefined,
                    Object {
                      "fontWeight": "bold",
                    },
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Sign Up
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
